File: Java/Petshop.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

// Class Petshop untuk merepresentasikan produk di petshop
class Petshop {
    private int id;
    private String nama;
    private String kategori;
    private int harga;
    private int stok;
    private static List<Petshop> data = new ArrayList<>(); // List untuk menyimpan data produk

    // Constructor nilai default
    public Petshop() {
        this.id = 0;
        this.nama = "";
        this.kategori = "";
        this.harga = 0;
        this.stok = 0;
    }

    // Constructor parameter atribut
    public Petshop(int id, String nama, String kategori, int harga, int stok) {
        this.id = id;
        this.nama = nama;
        this.kategori = kategori;
        this.harga = harga;
        this.stok = stok;
    }

    // Getter dan Setter untuk setiap atribut
    // id
    public int get_id() { return id; }
    public void set_id(int id) { this.id = id; }

    // nama
    public String get_nama() { return nama; }
    public void set_nama(String nama) { this.nama = nama; }

    // kategori
    public String get_kategori() { return kategori; }
    public void set_kategori(String kategori) { this.kategori = kategori; }

    // harga
    public int get_harga() { return harga; }
    public void set_harga(int harga) { this.harga = harga; }

    // stok
    public int get_stok() { return stok; }
    public void set_stok(int stok) { this.stok = stok; }

    // METHOD-METHOD YANG DIPERLUKAN UNTUK CRUD
    // Method untuk menampilkan informasi produk
    public static void tampilkanInfo(Petshop pet) {
        System.out.println("ID: " + pet.get_id());
        System.out.println("Nama Produk: " + pet.get_nama());
        System.out.println("Kategori: " + pet.get_kategori());
        System.out.println("Harga: " + pet.get_harga());
        System.out.println("Stok: " + pet.get_stok());
        System.out.println("-------------------------");
    }

    // Method untuk menambahkan data produk baru
    public static void create() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID: ");
        int id = scanner.nextInt();
        scanner.nextLine();  // Consume newline

        // Cek apakah ID sudah ada
        for (Petshop pet : data) {
            if (pet.get_id() == id) {
                System.out.println("ID sudah ada. Silakan masukkan ID yang berbeda.");
                return;
            }
        }

        System.out.print("Masukkan Nama Produk: ");
        String nama = scanner.nextLine();
        System.out.print("Masukkan Kategori: ");
        String kategori = scanner.nextLine();
        System.out.print("Masukkan Harga: ");
        int harga = scanner.nextInt();
        System.out.print("Masukkan Stok: ");
        int stok = scanner.nextInt();

        data.add(new Petshop(id, nama, kategori, harga, stok));
        System.out.println("Data berhasil ditambahkan!");
    }

    // Method untuk menampilkan semua data produk
    public static void print() {
        if (data.isEmpty()) {
            System.out.println("Tidak ada data yang tersedia.");
        } else {
            System.out.println("-------------------------");
            System.out.println("Berikut Peralatan PetShop");
            System.out.println("-------------------------");
            for (Petshop pet : data) {
                tampilkanInfo(pet);
            }
        }
    }

    // Method untuk mengubah data produk berdasarkan ID
    public static void update() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID produk yang ingin diubah: ");
        int id = scanner.nextInt();
        scanner.nextLine();  // Consume newline

        boolean ditemukan = false;
        for (Petshop pet : data) {
            if (pet.get_id() == id) {
                System.out.print("Masukkan Nama Produk Baru: ");
                String nama = scanner.nextLine();
                System.out.print("Masukkan Kategori Baru: ");
                String kategori = scanner.nextLine();
                System.out.print("Masukkan Harga Baru: ");
                int harga = scanner.nextInt();
                System.out.print("Masukkan Stok Baru: ");
                int stok = scanner.nextInt();

                pet.set_nama(nama);
                pet.set_kategori(kategori);
                pet.set_harga(harga);
                pet.set_stok(stok);

                System.out.println("Data berhasil diubah!");
                ditemukan = true;
                break;
            }
        }

        if (!ditemukan) {
            System.out.println("Data dengan ID " + id + " tidak ditemukan.");
        }
    }

    // Method untuk menghapus data produk berdasarkan ID
    public static void delete() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID produk yang ingin dihapus: ");
        int id = scanner.nextInt();

        for (Petshop pet : data) {
            if (pet.get_id() == id) {
                data.remove(pet);
                System.out.println("Data berhasil dihapus!");
                return;
            }
        }

        System.out.println("Data dengan ID " + id + " tidak ditemukan.");
    }

    // Method untuk mencari data produk berdasarkan nama
    public static void search() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan Nama Produk yang ingin dicari: ");
        String nama = scanner.nextLine();

        boolean ditemukan = false;
        for (Petshop pet : data) {
            if (pet.get_nama().equals(nama)) {
                tampilkanInfo(pet);
                ditemukan = true;
            }
        }

        if (!ditemukan) {
            System.out.println("Data dengan nama " + nama + " tidak ditemukan.");
        }
    }
}

File: PHP/Petshop.php
<?php
class Petshop {
    // Properti untuk menyimpan data produk
    private int $id;
    private string $nama;
    private string $kategori;
    private int $harga;
    private int $stok;
    private string $photoUrl;
    private array $daftarProduk;

    // Constructor untuk inisialisasi objek Petshop
    public function __construct(int $id = 0, string $nama = "", string $kategori = "", int $harga = 0, int $stok = 0, string $photoUrl = "") {
        $this->id = $id;
        $this->nama = $nama;
        $this->kategori = $kategori;
        $this->harga = $harga;
        $this->stok = $stok;
        $this->photoUrl = $photoUrl;
        $this->daftarProduk = array(); // Inisialisasi array daftar produk
    }

    // Setter methods untuk mengubah nilai properti
    public function setId(int $id) {
        $this->id = $id;
    }

    public function setNama(string $nama) {
        $this->nama = $nama;
    }

    public function setKategori(string $kategori) {
        $this->kategori = $kategori;
    }

    public function setHarga(int $harga) {
        $this->harga = $harga;
    }

    public function setStok(int $stok) {
        $this->stok = $stok;
    }

    public function setPhotoUrl(string $photoUrl) {
        $this->photoUrl = $photoUrl;
    }

    // Getter methods untuk mengambil nilai properti
    public function ambilId() {
        return $this->id;
    }

    public function ambilNama() {
        return $this->nama;
    }

    public function ambilKategori() {
        return $this->kategori;
    }

    public function ambilHarga() {
        return $this->harga;
    }

    public function ambilStok() {
        return $this->stok;
    }

    public function ambilPhotoUrl() {
        return $this->photoUrl;
    }

    // CRUD Operations
    public function tambahProduk($id, $nama, $kategori, $harga, $stok, $photoUrl) {
        // Tambahkan produk baru jika id tidak ada yang sama
        $indexProdukDicari = $this->cariProduk($id);
        if ($indexProdukDicari == count($this->daftarProduk)) {
            $produkBaru = new Petshop($id, $nama, $kategori, $harga, $stok, $photoUrl);
            $this->daftarProduk[] = $produkBaru;
        } else {
            echo "ditemukan id sama\n";
        }
    }

    public function cariProduk(int $idProduk) {
        // Cari produk berdasarkan id
        $indeks = 0;
        $banyakProduk = count($this->daftarProduk);
        $ditemukan = false;
        while ($indeks < $banyakProduk && !$ditemukan) {
            if ($this->daftarProduk[$indeks]->ambilId() == $idProduk) {
                $ditemukan = true;
            } else {
                $indeks++;
            }
        }
        return $indeks;
    }

    public function ubahDataProduk($id, $nama, $kategori, $harga, $stok, $photoUrl) {
        // Ubah data produk jika id ditemukan
        $indexProdukDicari = $this->cariProduk($id);
        if ($indexProdukDicari != count($this->daftarProduk)) {
            $this->daftarProduk[$indexProdukDicari]->setNama($nama);
            $this->daftarProduk[$indexProdukDicari]->setKategori($kategori);
            $this->daftarProduk[$indexProdukDicari]->setHarga($harga);
            $this->daftarProduk[$indexProdukDicari]->setStok($stok);
            $this->daftarProduk[$indexProdukDicari]->setPhotoUrl($photoUrl);
        } else {
            echo "tidak ditemukan id sama\n";
        }
    }

    public function hapusDataProduk(int $idProduk) {
        // Hapus data produk jika id ditemukan
        $indexProdukDicari = $this->cariProduk($idProduk);
        if ($indexProdukDicari != count($this->daftarProduk)) {
            array_splice($this->daftarProduk, $indexProdukDicari, 1);
        } else {
            echo "tidak ditemukan id sama\n";
        }
    }

    // Method untuk mencari produk berdasarkan nama
    public function cariProdukByNama($nama) {
        $found = false;
        echo "<table>";
        echo "<tr>
                <th>ID</th>
                <th>Nama</th>
                <th>Kategori</th>
                <th>Harga</th>
                <th>Stok</th>
                <th>Photo</th>
            </tr>";
        
        foreach ($this->daftarProduk as $produk) {
            if (stripos($produk->ambilNama(), $nama) !== false) {
                echo "<tr>
                        <td>" . $produk->ambilId() . "</td>
                        <td>" . $produk->ambilNama() . "</td>
                        <td>" . $produk->ambilKategori() . "</td>
                        <td>" . $produk->ambilHarga() . "</td>
                        <td>" . $produk->ambilStok() . "</td>
                        <td><img src='" . $produk->ambilPhotoUrl() . "' width='100' height='100'/></td>
                    </tr>";
                $found = true;
            }
        }
        echo "</table>";
        
        if (!$found) {
            echo "<p>Tidak ada produk yang ditemukan dengan nama tersebut.</p>";
        }
    }

    // Method untuk menampilkan semua produk
    public function tampilkan() {
        echo "<table>";
        echo "<tr>
                <th>ID</th>
                <th>Nama</th>
                <th>Kategori</th>
                <th>Harga</th>
                <th>Stok</th>
                <th>Photo</th>
            </tr>";
        foreach ($this->daftarProduk as $produk) {
            echo "<tr>
                    <td>" . $produk->ambilId() . "</td>
                    <td>" . $produk->ambilNama() . "</td>
                    <td>" . $produk->ambilKategori() . "</td>
                    <td>" . $produk->ambilHarga() . "</td>
                    <td>" . $produk->ambilStok() . "</td>
                    <td><img src='" . $produk->ambilPhotoUrl() . "' width='100' height='100'/></td>
                </tr>";
        }
        echo "</table>";
    }
}
?>